sqld/src/replication/primary/logger.rs

```diff
@@ -863,6 +894,32 @@ impl ReplicationLogger {
     pub fn get_frame(&self, frame_no: FrameNo) -> Result<Frame, LogReadError> {
         self.log_file.read().frame(frame_no)
     }
+
+    pub fn maybe_compact(&self) -> anyhow::Result<bool> {
+        let mut log_file = self.log_file.write();
+        if !log_file.should_compact() {
+            // compaction is not necessary or impossible, so exit early
+            return Ok(false);
+        }
+
+        let last_frame = {
+            let mut frames_iter = log_file.rev_frames_iter()?;
+            let Some(last_frame_res) = frames_iter.next() else {
+                // the log file is empty, nothing to compact
+                return Ok(false)
+            };
+            last_frame_res?
+        };
+
+        let size_after = last_frame.header().size_after;
+        if size_after == 0 {
+            // last frame does not seem to be a commit boundary, so we can't do a compaction
+            return Ok(false);
+        }
```

Comment:
We can `assert` that: rev_iter's last frame must be a commit boundary since the iterator does not return uncommitted frames.

Reply:
Ok, changed